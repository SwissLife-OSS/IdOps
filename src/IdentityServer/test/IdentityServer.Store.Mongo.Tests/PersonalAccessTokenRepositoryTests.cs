using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using FluentAssertions;
using IdOps.IdentityServer.Model;
using Squadron;
using Xunit;

namespace IdOps.IdentityServer.Storage.Mongo.Tests;

[Collection(TestCollectionNames.Store)]
public class PersonalAccessTokenRepositoryTests : RepositoryTest
{
    public PersonalAccessTokenRepositoryTests(MongoResource mongoResource) : base(mongoResource) 
    { }


    [Fact]
    public async Task GetActiveTokensByUserNameAsync_UserNameHasDifferentCase_UserNameFound()
    {
        // Arrange
        var userName = "UserOne";
        var differentCaseUserName = "userone";
        var personalAccessToken = new IdOpsPersonalAccessToken
        {
            Id = Guid.NewGuid(),
            UserName = userName,
            Tokens = new List<IdOpsHashedToken>
            {
                new(
                    Guid.NewGuid(),
                    "foobar",
                    DateTime.Parse("2031.01.02"),
                    DateTime.Parse("2001.01.02"),
                    false)
            }
        };
        var repository = new PersonalAccessTokenRepository(DbContext);

        // Act
        await repository.CreateAsync(personalAccessToken, default);
        var activeTokens =
            await repository.GetActiveTokensByUserNameAsync(differentCaseUserName, default);

        // Assert
        activeTokens.Should().NotBeNullOrEmpty();
        activeTokens.Should().ContainSingle();
        activeTokens[0].UserName.Should().Be(userName);
    }
}
