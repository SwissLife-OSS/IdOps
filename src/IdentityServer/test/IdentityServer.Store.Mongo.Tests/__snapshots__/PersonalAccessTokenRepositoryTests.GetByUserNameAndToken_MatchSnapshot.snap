{
  "Id": "19d7b959-aa75-422b-a675-7185b44b4be5",
  "UserName": "userName",
  "Token": "tokensecret",
  "CreatedAt": "2020-12-09T11:36:58.832Z",
  "ExpiresAt": "2021-12-09T11:36:58.833Z",
  "IsOneTime": false,
  "IsUsed": false,
  "AllowedScopes": [
    "api1"
  ],
  "AllowedClients": [
    "Test.PAT01"
  ]
}
